Utilities/DataLoading/EdFi.LoadTools/Engine/FileImportPipeline/FileImportPipeline.cs

```diff
@@ -0,0 +1,90 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Linq;
+using EdFi.LoadTools.ApiClient;
+
+namespace EdFi.LoadTools.Engine.FileImportPipeline
+{
+    public class FileImportPipeline
+    {
+        private readonly IFileImportPipelineStep[] _steps;
+        private readonly IFileContextProvider _fileContextProvider;
+        private readonly IXmlReferenceCacheFactory _xmlReferenceCacheFactory;
+
+        public FileImportPipeline(IXmlReferenceCacheFactory xmlReferenceCacheFactory,
+            IFileImportPipelineStep[] steps,
+            IFileContextProvider fileContextProvider)
+        {
+            _xmlReferenceCacheFactory = xmlReferenceCacheFactory;
+            _steps = steps;
+            _fileContextProvider = fileContextProvider;
+        }
+
+        public IEnumerable<ApiLoaderWorkItem> RetrieveResourcesFromFiles(List<string> resources, int level)
+        {
+            foreach (var fileContext in _fileContextProvider.GetFileContexts())
```

Comment:
We lost the filter here.  We don't need the concept of interchange grouping anymore since everything is resource based, but we should still only be processing files here based on the provided list of resources.  Otherwise we're going to stream through a bunch of files we don't have any interest in.

Reply:
fixed